src/Query/QueryBuilder.php

```diff
@@ -1625,4 +1632,23 @@ public function disableResultCache(): self
 
         return $this;
     }
+
+    public function withComment(string $comment): self
```

Comment:
This `with` prefix might be misleading. A wither typically returns a new instance of the same object, with the modifications, without altering the existing instance. Here, you are altering the existing instance and you are not cloning anything.

```suggestion
    public function addComment(string $comment): self
```